docs/live-streaming.md

```diff
@@ -0,0 +1,157 @@
+
+
+Live Streaming Process
+======================
+
+Now that you know how the Live Streaming should turn a bunch of MPEG-TS
+encoded video stream into a network live stream, let's complete the definition
+of the live streaming transport over the network using SRT.
+
+The source side should be a real live stream, that is, for example:
+
+- a file read by an application that can make a live stream out of it
+- a grab device that is grabbing the data in constant rates and then
+the data are passed through the live encoder
+- some existing live network encoder that streams a live stream over UDP
+- a camera that uses some simple encoding method, like MJPEG (needs transcoding)
+
+As an example of an application that can make a live stream out of a file
+can be `ffmpeg`. You just need to remember about two things here:
+
+- The `-re` options is required for making the live stream out of a file
+- The `pkt_size=1316` parameter should be added to UDP output URI, if you make
+ffmpeg output to UDP
+
+Pay attention first that the splitting the data for transport into single units
+that can fit in one UDP packet, and first of all defining time distances
+between these units is something that must come in synch with the timestamps
+in the transport stream. Even if you have a grab device, it usually will make
+just one video frame plus some audio at specified time, but still this must
+be split into single network units with appropriate time distance between
+them. Still, this can only be done by an application, which well knows the
+type of the stream and knows how it must be most properly split into the
+time-divided single network transport units.
+
+The `stransmit` application, or any other application that uses SRT for
+reading, should read always in 1316-byte portions (network transport units) and
+feed each such unit into the call to appropriate `srt_send*` function. The
+important part of that process is that these 1316 units appear at exactly
+required times so that SRT can replay them with the identical time distance
+between them at the reception side.
+
+So, you feed these 1316-byte units into SRT, SRT will send them to the
+other side, applying a configurable delay, known as "latency". This is an
+extra time that the packet will have to spend in the "anteroom" on the
+receiving side before it's delivered to the output. This time should cover
+both the unexpectedly delayed transmission of one UDP packet, as well
+as have extra time for a case when the packet was lost and had to be
+retransmitted. Every UDP packet carrying an SRT packet has a timestamp,
+which is grabbed at the time when the packet is given up to SRT for
+sending, and using that timestamp the appropriate delay is applied before
+delivering to the output, so that time distances between two consecutive
+packets at the delivery application is aimed to be exactly the same as
+the time distance between these same packets at the time when they are
+given up to SRT for sending.
```

Comment:
given to SRT

Reply:
passed to SRT should sound better anyway